results-processor/wptreport.py

```diff
@@ -481,7 +481,27 @@ def prepare_labels(report, labels_str, uploader):
     # Remove any empty labels.
     if '' in labels:
         labels.remove('')
-    return sorted(labels)
+    return labels
+
+
+def normalize_product(report):
+    """Normalizes the product identifier.
+
+    Computes what labels need to be added while normalizing the product.
+
+    Args:
+        report: A WPTReport
+
+    Returns:
+       A set of strings.
+    """
+    product = report.run_info['product']
+    if "_" in product:
+        tokens = product.split("_")
+        report.run_info['product'] = tokens[0]
+        return set(tokens[1:])
```

Comment:
I'm wondering whether labelling as the full original product string is a better idea here (e.g. `edge_webdriver`) to avoid reusing labels that might have other meaning. A little paranoid, maybe.

@foolip - thoughts?

Reply:
Given that a URL like https://wpt.fyi/results/?label=edge&label=master doesn't redirect and people might assume it to be the same as https://wpt.fyi/results/?label=master&product=edge, the most conservative and also non-lossy option seems to be labeling with both "edge" and "edge_webdriver" if the product was "edge_webdriver".

Reply:
@lukebjerring note that because this is called after `prepare_labels` we end up with both `edge` and `edge_webdriver` in this case.